File: client/src/app/_models/academicInformation.ts
import { Member } from "./member";

export interface AcademicInformation {
    id: number;
    appUser: Member;
    appUserId: number;
    currentDegreeProgram: string;
    yearOfStudy: string;
    expectedGraduationDate: string;
    overallGPA: number;
    majorGPA: number;
    completedCourses: string; // JSON string to store list of completed courses
    currentCourses: string;   // JSON string to store list of current courses
    favoriteSubjects: string; // JSON string to store list of favorite subjects
    leastFavoriteSubjects: string; // JSON string to store list of least favorite subjects
    technicalSkills: string;  // JSON string to store list of technical skills
    softSkills: string;       // JSON string to store list of soft skills
    certifications: string;   // JSON string to store list of certifications
    projects: string;         // JSON string to store list of projects
    internships: string;      // JSON string to store list of internships
    clubsAndOrganizations: string; // JSON string to store list of clubs and organizations
    volunteerExperience: string; // JSON string to store list of volunteer experience
    feedbackOnCurriculum: string;
  }

File: API/Entities/AcademicInformation.cs
using System;
using System.Collections.Generic;
using Microsoft.AspNetCore.Identity;

namespace API.Entities
{
    public class AcademicInformation
    {
        public int Id { get; set; }
        public AppUser appUser  { get; set; }
        public int AppUserId { get; set; }
        public string CurrentDegreeProgram { get; set; }
        public string YearOfStudy { get; set; }
        public DateOnly ExpectedGraduationDate { get; set; } = DateOnly.FromDateTime(DateTime.Now);
        public double OverallGPA { get; set; }
        public double MajorGPA { get; set; }
        public string CompletedCourses { get; set; }  // JSON string to store list of completed courses
        public string CurrentCourses { get; set; }    // JSON string to store list of current courses
        public string FavoriteSubjects { get; set; }    // JSON string to store list of current courses
        public string LeastFavoriteSubjects { get; set; }    // JSON string to store list of current courses
        public string TechnicalSkills { get; set; }    // JSON string to store list of current courses
        public string SoftSkills { get; set; }    // JSON string to store list of current courses
        public string Certifications { get; set; }    // JSON string to store list of current courses
        public string Projects { get; set; }    // JSON string to store list of current courses
        public string Internships { get; set; }    // JSON string to store list of current courses
        public string ClubsAndOrganizations { get; set; }    // JSON string to store list of current courses
        public string VolunteerExperience { get; set; }    // JSON string to store list of current courses
        public string FeedbackOnCurriculum { get; set; }

    }
}